core/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -100,11 +100,11 @@
     static {
         Function<Settings, String> defValue = settings -> {
             // TODO: uncomment this
-            /*boolean singleType = true;
+            boolean singleType = true;
             if (settings.getAsVersion(IndexMetaData.SETTING_VERSION_CREATED, null) != null) {
                 singleType = Version.indexCreated(settings).onOrAfter(Version.V_6_0_0_alpha1_UNRELEASED);
-            }*/
-            boolean singleType = false;
+            }
+            //boolean singleType = false;
```

Comment:
I just wanted to make indices require one type at most by default for testing, I will merge this change after #24428.